File: app/src/main/java/org/neidhardt/dynamicsoundboard/soundsheetmanagement/tasks/LoadSoundSheetsTask.java
package org.neidhardt.dynamicsoundboard.soundsheetmanagement.tasks;

import de.greenrobot.event.EventBus;
import org.neidhardt.dynamicsoundboard.dao.DaoSession;
import org.neidhardt.dynamicsoundboard.dao.SoundSheet;
import org.neidhardt.dynamicsoundboard.misc.longtermtask.LongTermTask;
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.events.SoundSheetsLoadedEvent;
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataStorage;

import java.util.List;

/**
 * File created by eric.neidhardt on 08.05.2015.
 */
public class LoadSoundSheetsTask extends LongTermTask<List<SoundSheet>>
{
	private static final String TAG = LoadSoundSheetsTask.class.getName();

	private DaoSession daoSession;
	private SoundSheetsDataStorage soundSheetsDataStorage;

	public LoadSoundSheetsTask(DaoSession daoSession, SoundSheetsDataStorage soundSheetsDataStorage)
	{
		this.daoSession = daoSession;
		this.soundSheetsDataStorage = soundSheetsDataStorage;
	}

	@Override
	public List<SoundSheet> call() throws Exception
	{
		return this.daoSession.getSoundSheetDao().queryBuilder().list();
	}

	@Override
	protected void onSuccess(List<SoundSheet> loadedSoundSheets) throws Exception
	{
		super.onSuccess(loadedSoundSheets);
		this.soundSheetsDataStorage.addLoadedSoundSheets(loadedSoundSheets);
	}

	@Override
	protected String getTag()
	{
		return TAG;
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/soundsheetmanagement/tasks/LoadSoundSheetsTask.kt
package org.neidhardt.dynamicsoundboard.soundsheetmanagement.tasks

import org.neidhardt.dynamicsoundboard.dao.DaoSession
import org.neidhardt.dynamicsoundboard.dao.SoundSheet
import org.neidhardt.dynamicsoundboard.longtermtask.LongTermTask
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataStorage

/**
 * File created by eric.neidhardt on 08.05.2015.
 */
public class LoadSoundSheetsTask
(
		private val daoSession: DaoSession,
		private val soundSheetsDataStorage: SoundSheetsDataStorage
)
: LongTermTask<List<SoundSheet>>()
{
	override val TAG = javaClass.name

	@Throws(Exception::class)
	override fun call(): List<SoundSheet>
	{
		return this.daoSession.soundSheetDao.queryBuilder().list()
	}

	@Throws(Exception::class)
	override fun onSuccess(result: List<SoundSheet>)
	{
		super.onSuccess(result)
		for (soundSheet in result)
			this.soundSheetsDataStorage.addSoundSheetToManager(soundSheet)
	}


}
